scripts/tuple_tree_generator/tuple_tree_generator/templates/tuple_tree_gen.ts.tpl

```diff
@@ -119,7 +119,7 @@ export /** if class_.abstract **/abstract/** endif **/ class /*= class_.name =*/
         /**- endif **/
         /**- if class_.inherits **/
         /**- if class_.inherits | get_guid **/
-        super(rawObject, gen/*= class_.name =*/Guid);
+        super(rawObject, genGuid);
```

Comment:
I'm not sure what you're talking about. I didn't look too deeply into how GUIDs are used: all I know is that primitive types no longer need to be a special little unicorn, dropping other related infrastructure can always come later.

Reply:
1. Before the switch to incremental type IDs, we were using random numbers are IDs, i.e., GUIDs. we no longer do, so it'd make sense to drop all references to "guid". but yeah, that can come later.
2. Side question. What's the first ID we use now? `0`? `1`?

Reply:
1. Yeah, I tried to limit changes to areas I'm not comfortable with. This is already way too big.
2. Both make sense, I went with `0` since that's what I'm used to counting from. Downside is that we lost the "0 means no ID was assigned convention", but consider that we are going to have to extend TTG with default value support anyway (if we intend to keep any kinds of configs in TTG-generated data-structures that is) - at that point we can assign a new, more natural "no ID" value, like, say, `-1` and be happy.